src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -113,9 +113,31 @@ MyDiagnoses.navigationOptions = () => ({
   header: null
 })
 
-const MyDiagnosesStack = createStackNavigator({
-  MyDiagnoses: MyDiagnoses,
-  DetailedDiagnose: DetailedDiagnose
-})
+const MyDiagnosesStack = buildStackNavigator(
+  {
+    MyDiagnoses,
+    DetailedDiagnose
+  },
+  {
+    initialRouteName: 'MyDiagnoses'
+  }
+)
+
+MyDiagnosesStack.navigationOptions = ({ navigation }) => {
+  let tabBarVisible
+  const allowedRoutesToShowTabBar = ['MyDiagnoses']
+
+  if (navigation.state.routes.length > 1) {
+    navigation.state.routes.map(route => {
+      if (includes(allowedRoutesToShowTabBar, route.routeName)) {
+        tabBarVisible = true
+      } else {
+        tabBarVisible = false
+      }
+    })
+  }
+
+  return { tabBarVisible }
+}
```

Comment:
- The constant `allowedRoutesToShowTabBar` should be in the top-level, otherwise you'd be creating a new array on each call unnecessarily.
- Never use `map` to iterate over an array, it's meant to **modify** the contents of the array. Use `forEach` for iterating, it doesn't create a new array.
- This code can be written in a much shorter and readable way. I think it's the same behavior?
```javascript
import { isEmpty } from 'lodash'
// ...
const allowedRoutesToShowTabBar = ['MyDiagnoses']
// ...
MyDiagnosesStack.navigationOptions = ({ navigation }) => {
  if (navigation.state.routes.length <= 1) return {}

  const routesInHistoryWithTabBar = intersect(navigation.state.routes, allowedRoutesToShowTabBar)
  
  return { tabBarVisible: !isEmpty(routesInHistoryWithTabBar) }
}
```

Reply:
I changed the code to reflect this. This should be executed once though. It's a change in a static property of the component. Nevertheless, once we stop modifying this component to add new features we should move this into a separate module like the one in `DetailedDiagnose`